xblockutils/studio_editable.py

```diff
@@ -0,0 +1,295 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015 OpenCraft
+# License: AGPLv3
+"""
+This module contains a mixin that allows third party XBlocks to be easily edited within edX
+Studio just like the built-in modules. No configuration required, just add
+StudioEditableXBlockMixin to your XBlock.
+"""
+
+# Imports ###########################################################
+
+import json
+import logging
+
+from xblock.core import XBlock
+from xblock.fields import Scope, JSONField, List, Integer, Float, Boolean, String
+from xblock.exceptions import JsonHandlerError
+from xblock.fragment import Fragment
+from xblock.validation import Validation
+
+from xblockutils.resources import ResourceLoader
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+# Classes ###########################################################
+
+
+class FutureFields(object):
+    """
+    A helper class whose attribute values come from the specified dictionary or fallback object.
+    """
+    def __init__(self, new_fields_dict, newly_removed_fields, fallback_obj):
```

Comment:
I'd like to see a docstring which explains what these three parameters represent